src/test/java/PageFactory/BaseStepPage.java

```diff
@@ -0,0 +1,20 @@
+package PageFactory;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.PageFactory;
+
+public abstract class BaseStepPage {
+    protected WebDriver driver;
+    protected String url;
+
+    public BaseStepPage(WebDriver driver) {
```

Comment:
No needs to pass WebDriver instance as parameter. You can access it via singleton class